### PR TITLE
messaging: simplified -> Breaking Change

### LINKED ISSUE #5947: RFC: Messager simplification -> Breaking change
After the move to RBR for messaging, and discussions with @derekperkins, we're simplifying the messaging API. On the upside, this will support all insert constructs. On the downside, the application has more responsibilities on correctness and features. What is changing:
* time_created and time_scheduled are removed from the list of mandatory columns. They will neither be checked nor be populated. If necessary, the application can add these columns and keep them populated.
* The primary key that used to be (time_scheduled,id) is now just id.
* The unique key on id is not required any more, since the PK covers that.
* time_next remains a nullable column, but it's recommended to default it to 0. Setting it to 0 (or any value < now) will schedule the message for immediate delivery. Previously a 0-value was treated as an acked message. This is not the case any more. This means that one could insert a row without specifying a time_next. The code checks for time_acked to contain a non-null (and non-zero) value to determine that a message was acked.
* The default value for epoch can also be 0, but the code treats null values as 0 also.
* Purging is by time_acked. Consequently, a new index is needed on time_acked.

Here's an idiomatic message table definition:
```
var createMessage = `create table vitess_message(
	id bigint,
	time_next bigint default 0,
	epoch bigint,
	time_acked bigint,
	message varchar(128),
	primary key(id),
	index next_idx(time_next, epoch),
	index ack_idx(time_acked))
comment 'vitess_message,vt_ack_wait=30,vt_purge_after=120,vt_batch_size=1,vt_cache_size=1000,vt_poller_interval=30'
```

Here's an idiomatic insert:
```
insert into vitess_message(id, message) values(1, 'hello world')
```

### PR DESCRIPTION
**Pull request by sougou**
Closes #5947 

Details are also covered in the issue.

### CODE REVIEW

File: go/vt/vttablet/tabletserver/planbuilder/builder.go [hunk 1]
```diff
@@ -73,9 +74,10 @@ func analyzeUpdate(upd *sqlparser.Update, tables map[string]*schema.Table) (plan
 		plan.WhereClause = buf.ParsedQuery()
 	}
 
-	// If plan.Table==nil, it's likely a multi-table statement.
-	// MySQL doesn't allow limit clauses for multi-table dmls.
-	// If there's an explicity Limit, honor it.
+	// Situations when we pass-through:
+	// PassthroughDMLs flag is set.
+	// plan.Table==nil: it's likely a multi-table statement. MySQL doesn't allow limit clauses for multi-table dmls.
+	// If there's an explicity Limit.
```
Comment (derekperkins) [inline go/vt/vttablet/tabletserver/planbuilder/builder.go]:
```suggestion
	// If there's an explicit Limit.
```

File: go/test/endtoend/messaging/main_test.go [hunk 2]
```diff
@@ -41,28 +41,24 @@ var (
 	userKeyspace         = "user"
 	lookupKeyspace       = "lookup"
 	createShardedMessage = `create table sharded_message(
-		time_scheduled bigint,
 		id bigint,
-		time_next bigint,
+		time_next bigint default 0,
 		epoch bigint,
```
Comment (derekperkins) [inline go/test/endtoend/messaging/main_test.go]: Why not default epoch too?

Comment (sougou) [reply go/test/endtoend/messaging/main_test.go]: This is to demonstrate that it works with no default.

**Comment by sougou** [comment]
github commit suggestion phailed. It doesn't add DCO...
I'll repush a fixed version.

**Pull request merged**